Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5402,6 +5386,46 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
         }
 
 
+        /// <summary>
+        /// Normalise crank angle so that it is a value between 0 and 360 starting at the real crank angle difference
+        /// </summary>
+        private float NormalisedCrankAngle( int cylinderNumber)
```

Comment:
This whole function can be simplified to
```csharp
float crankPositionRad = LocomotiveAxles[0].AxlePositionRad + WheelCrankAngleDiffRad[cylinderNumber];
return (float)(MathHelper.WrapAngle(crankPositionRad)+Math.PI) // Range angle from 0 to 360º
```

Reply:
I have tried the suggested changes and it doesn't line up with the current values that I am calculating.

Reply:
Obviously not, sorry. You can keep your code

Reply:
Obviously not, sorry. This is my proposal, to reduce number of operations:
```csharp
float normalisedCrankAngleRad = (float)MathHelper.WrapAngle(LocomotiveAxles[0].AxlePositionRad+WheelCrankAngleDiffRad[cylinderNumber]);
if (normalisedCrankAngleRad < 0) normalisedCrankAngleRad += (float)(2*Math.PI);
return normalisedCrankAngleRad;
```